Dockerfile-dev

```diff
@@ -1,36 +0,0 @@
-FROM node:20.11.1-bookworm-slim
```

Comment:
we no longer encourage docker-based local development for performance and simplicity reasons

Reply:
@kibertoad I guess we need to update readme

Reply:
I think I've removed all docker references from it, but please let me know if I missed any